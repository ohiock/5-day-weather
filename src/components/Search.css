.container {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  margin: auto;
  width: 80%;
}

.container div {
  padding: 0 0 15% 0;
}

.city-textbox, .go-button {
  padding: 0 2vw;
  width: 80%;

  background: #f1f1f1;
  border: 0;
  color: #888888;
  font-size: 5.5vw;
}

.go-button {
  width: 20%;

  background: #888888;
  color: #f1f1f1;
}

.go-button:hover {
  background: #8f8f8f;
}

.go-button:active {
  background: #818181;
}

.call-to-action {
  line-height: 1.15;
  margin-bottom: 2vw;

  font-size: 6.5vw;
}

@media (max-width: 768px) {
  .city-textbox, .go-button {
    padding: 1.5vw 3.5vw;
  }
}
