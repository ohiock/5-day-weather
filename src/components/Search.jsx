import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CSSModules from 'react-css-modules';

import Config from '../constants/Config';
import ErrorMessages from '../constants/ErrorMessages';
import ErrorMessage from './ErrorMessage';

import styles from './Search.css';

const propTypes = {
  searchCallback: PropTypes.func.isRequired,
};

class Search extends Component {
  constructor(props) {
    super(props);

    this.state = {
      city: '',
      inputError: '',
    };

    this.onChangeCityTextbox = this.onChangeCityTextbox.bind(this);
    this.onClickGoButton = this.onClickGoButton.bind(this);
  }

  isValidInput(input) {
    const regExResult = Config.CITY_INPUT_REGEX.test(input);
    return input.length <= Config.CITY_INPUT_CHARACTER_LENGTH && regExResult;
  }

  onChangeCityTextbox(event) {
    const input = event.target.value.trim();

    if (!this.isValidInput(input)) {
      this.setState({ inputError: ErrorMessages.INVALID_CITY_INPUT });

      return;
    }

    this.setState({ city: input, inputError: '' });
  }

  onClickGoButton() {
    if (this.state.inputError) {
      return;
    }

    this.props.searchCallback(this.state.city);
  }

  render() {
    return (
      <div styleName="container">
        <div>
          <h1 styleName="call-to-action">Give me the weather<br /> forecast for</h1>
          <ErrorMessage show={!!this.state.inputError} errorMessage={this.state.inputError} />
          <input onChange={this.onChangeCityTextbox} type="text" placeholder="Enter a city" styleName="city-textbox" />
          <button onClick={this.onClickGoButton} styleName="go-button">GO</button>
        </div>
      </div>
    );
  }
}

Search.propTypes = propTypes;

export default CSSModules(Search, styles);
