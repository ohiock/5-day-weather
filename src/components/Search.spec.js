import React from 'react';
import renderer from 'react-test-renderer';

import Search from './Search';

it('renders correctly', () => {
  const tree = renderer.create(
    <Search searchCallback={() => true} show={true} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
