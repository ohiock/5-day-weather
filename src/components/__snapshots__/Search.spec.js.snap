// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="undefined undefined"
>
  <div>
    <h1
      className={undefined}
    >
      Give me the
      <br />
       weather forecast for
    </h1>
    <p
      className="undefined undefined"
    >
      Please enter a valid city. It must be composed of only alphabetical letters and less than 50 chracters in length.
    </p>
    <input
      className={undefined}
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Enter a city"
      type="text"
    />
    <button
      className={undefined}
      onClick={[Function]}
    >
      GO
    </button>
  </div>
</div>
`;
